### PR TITLE
react grid ui for v4

### PR DESCRIPTION
**Pull request by phanirithvij**
### Description

The updated grid UI with status filtering and status updates. 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Tests are not fully done. will be adding them in a followup PR.

### CODE REVIEW

File: java/server/src/org/openqa/selenium/grid/console/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
+# testing
+/coverage
+
```
Comment (AutomatedTester) [inline java/server/src/org/openqa/selenium/grid/console/.gitignore:10]: Please add these to the repo `.gitignore`

File: java/server/src/org/openqa/selenium/grid/console/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "useTabs": true,
+    "jsxBracketSameLine": false,
+    "endOfLine": "auto"
+}
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/.prettierrc]: Please leave a line at EOF.

File: java/server/src/org/openqa/selenium/grid/console/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,15811 @@
+{
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/package-lock.json:1]: I think this file is not required and can be removed.

File: java/server/src/org/openqa/selenium/grid/console/public/css/bootstrap.min.css [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/*!
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/public/css/bootstrap.min.css:1]: Should this file be saved or directly fetched over the internet? @AutomatedTester  what do you think?

File: java/server/src/org/openqa/selenium/grid/console/public/css/fontawesome.min.css [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/*!
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/public/css/fontawesome.min.css:1]: Same for this.

File: java/server/src/org/openqa/selenium/grid/console/public/index.html [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  <meta charset="utf-8" />
+  <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta name="theme-color" content="#000000" />
+  <meta name="description" content="Web site created using create-react-app" />
+  <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+  <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+  <link rel="stylesheet" href="%PUBLIC_URL%/css/bootstrap.min.css">
+  <link rel="stylesheet" href="%PUBLIC_URL%/css/fontawesome.min.css">
+  <!-- <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet"> -->
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/public/index.html]: Remove comment

File: java/server/src/org/openqa/selenium/grid/console/src/App.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+
+// css import order is important
+/* 1 */ import "./css/theme.css";
+/* 2 */ import "./css/theme-selenium.css";
+/* 3 */ import "./App.css";
+
+import HelpPage from "./screens/HelpPage/HelpPage";
+import Console from "./screens/Console/Console";
+// import SearchAppBar from "./components/AppBar/AppBar";
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/App.tsx]: Unnecessary comment

File: java/server/src/org/openqa/selenium/grid/console/src/assets/chrome.svg [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        <stop offset=".954" stop-color="#f7390a"/>
+        <stop offset="1" stop-color="#f73100"/>
+    </linearGradient>
+    <path fill="url(#Pax8JcnMzivu8f~SZ~k1yc)" d="M41.84,15H24c-4.97,0-9,4.03-9,9c0,1.49,0.36,2.89,1.01,4.13H16L7.16,13.26H7.14 C10.68,7.69,16.91,4,24,4C31.8,4,38.55,8.48,41.84,15z"/>
+    <linearGradient id="Pax8JcnMzivu8f~SZ~k1yd" x1="19.05" x2="28.95" y1="30.95" y2="21.05" gradientTransform="matrix(1 0 0 -1 0 50)" gradientUnits="userSpaceOnUse">
+        <stop offset="0" stop-color="#2aa4f4"/>
+        <stop offset="1" stop-color="#007ad9"/>
+    </linearGradient>
+    <path fill="url(#Pax8JcnMzivu8f~SZ~k1yd)" d="M31,24c0,3.867-3.133,7-7,7s-7-3.133-7-7s3.133-7,7-7S31,20.133,31,24z"/>
+</svg>
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/assets/chrome.svg]: Leave line after EOF.

File: java/server/src/org/openqa/selenium/grid/console/src/assets/edge-new.svg [hunk 9]
```diff
@@ -0,0 +1 @@
...th fill="url(#2pm2zgLwrfHW4PMfqfOjYb)" d="M18.985,23.894c0-2.298,1.775-4.575,4.395-4.871c-4.104,0.349-8.798,4.375-8.798,11.831	c0,7.963,6.864,13.465,13.943,12.628c8.386-1.927,12.441-8.74,12.441-9.168c0-0.523-0.451-0.603-0.647-0.603	c-0.643,0-2.444,1.828-7.403,1.828C24.615,35.539,18.985,29.847,18.985,23.894z"/><radialGradient id="2pm2zgLwrfHW4PMfqfOjYc" cx="47.648" cy="16.079" r="61.532" gradientTransform="matrix(-1 0 0 -1.5084 95.296 40.31)" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#5eeb69"/><stop offset=".214" stop-color="#4bc958"/><stop offset=".225" stop-color="#4ac862"/><stop offset=".272" stop-color="#47c68b"/><stop offset=".323" stop-color="#44c4ae"/><stop offset=".376" stop-color="#41c2cb"/><stop offset=".433" stop-color="#3fc1e1"/><stop offset=".495" stop-color="#3ec0f1"/><stop offset=".565" stop-color="#3dbffa"/><stop offset=".664" stop-color="#3dbffd"/></radialGradient><path fill="url(#2pm2zgLwrfHW4PMfqfOjYc)" d="M4,23.481C4,15.673,10.557,4,24.014,4C37.452,4,44,13.923,44,20.905	c0,5.604-4.429,9.732-9.992,9.732c-4.245,0-6.759-1.408-6.759-2.53c0-0.726,1.755-1.837,1.755-3.982	c0-2.565-2.076-10.678-12.628-10.678C13.442,13.448,5.256,14.745,4,23.481z"/></svg>
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/assets/edge-new.svg]: Have you used an SVG compressor?

File: java/server/src/org/openqa/selenium/grid/console/src/assets/firefox.svg [hunk 10]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/assets/firefox.svg]: It's better to use SVG compressor here. Thoughts @AutomatedTester ?

File: java/server/src/org/openqa/selenium/grid/console/src/components/KeyBoard/Keyboard.css [hunk 11]
```diff
@@ -0,0 +1,8 @@
+/* This is to prevent the rings with z-index from going on top of the modal */
+body>div.ReactModalPortal>div {
+  z-index: 2;
+}
+
+body > div.ReactModalPortal > div > div{
+  background: rgb(231, 255, 235) !important;
+}
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/components/KeyBoard/Keyboard.css]: Empty line at EOF.

File: java/server/src/org/openqa/selenium/grid/console/src/components/Node/Node.module.css [hunk 12]
```diff
@@ -0,0 +1,3 @@
+.tdnoborder{
+  border: 0 !important;
+}
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/components/Node/Node.module.css]: And here as well.

File: java/server/src/org/openqa/selenium/grid/console/src/css/theme-selenium.css [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+    font-size: 1rem !important;
+}
+
+/*
+  Color for the select element of the languages
+*/
+
+.select-style select option {
+    color: black;
+}
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/css/theme-selenium.css]: And here as well :)

File: java/server/src/org/openqa/selenium/grid/console/src/css/theme.css [hunk 14]
```diff
@@ -0,0 +1,1179 @@
...
+    font-style: normal;
+    font-weight: 300;
+    src: url("../fonts/Work_Sans_300.eot?#iefix") format("embedded-opentype"), url("../fonts/Work_Sans_300.woff") format("woff"), url("../fonts/Work_Sans_300.woff2") format("woff2"), url("../fonts/Work_Sans_300.svg#WorkSans") format("svg"), url("../fonts/Work_Sans_300.ttf") format("truetype");
+}
+@font-face {
+    font-family: 'Work Sans';
+    font-style: normal;
+    font-weight: 500;
+    src: url("../fonts/Work_Sans_500.eot?#iefix") format("embedded-opentype"), url("../fonts/Work_Sans_500.woff") format("woff"), url("../fonts/Work_Sans_500.woff2") format("woff2"), url("../fonts/Work_Sans_500.svg#WorkSans") format("svg"), url("../fonts/Work_Sans_500.ttf") format("truetype");
+} */
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/css/theme.css]: Remove unnecessary comments

File: java/server/src/org/openqa/selenium/grid/console/src/react-app-env.d.ts [hunk 15]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/react-app-env.d.ts:1]: Is this file required?

File: java/server/src/org/openqa/selenium/grid/console/src/screens/HelpPage/Help.module.css [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+    color: whitesmoke !important;
+}
+
+.inline {
+    display: inline-block !important;
+}
+
+.header {
+    height: 8rem;
+}
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/screens/HelpPage/Help.module.css]: And here.

File: java/server/src/org/openqa/selenium/grid/console/src/screens/Hub/Hub.css [hunk 17]
```diff
@@ -0,0 +1,10 @@
+.progressbar-text {
+  top: 50% !important;
+  letter-spacing: 8px;
+}
+
+#container {
+  position: absolute;
+  /* position: absolute !important; */
+  top: 50%;
+}
```
Comment (shucon) [inline java/server/src/org/openqa/selenium/grid/console/src/screens/Hub/Hub.css]: And here

**Comment by shucon** [comment]
Feel free to resolve any comment that's not relevant.

File: java/server/src/org/openqa/selenium/grid/console/public/index.html [hunk 18]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8" />
+  <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta name="theme-color" content="#000000" />
+  <meta name="description" content="Web site created using create-react-app" />
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/console/public/index.html]: I don't think we need this.

File: java/server/src/org/openqa/selenium/grid/console/public/index.html [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+  <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (shs96c) [inline java/server/src/org/openqa/selenium/grid/console/public/index.html:8]: This should be the selenium logo, not the React one.

File: java/server/src/org/openqa/selenium/grid/console/src/react-app-env.d.ts (hunk 15 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/src/react-app-env.d.ts:1]: Yep. React needs it when compiling.

File: java/server/src/org/openqa/selenium/grid/console/package-lock.json (hunk 3 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/package-lock.json:1]: https://stackoverflow.com/a/44210813/8608146 it should be checked in source control?

File: java/server/src/org/openqa/selenium/grid/console/public/css/bootstrap.min.css (hunk 4 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/public/css/bootstrap.min.css:1]: David said the website should work offline (in an intranet) that's why I've added it in the project.

File: java/server/src/org/openqa/selenium/grid/console/public/css/fontawesome.min.css (hunk 5 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/public/css/fontawesome.min.css:1]: same for this.

File: java/server/src/org/openqa/selenium/grid/console/src/assets/chrome.svg (hunk 8 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/src/assets/chrome.svg]: I don't think it matters for svg asset files. I didn't write those contents manually.

File: java/server/src/org/openqa/selenium/grid/console/src/assets/edge-new.svg (hunk 9 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/src/assets/edge-new.svg]: Yeah. I did minify each one of the svgs by importing them in Adobe XD and exporting them as Optimized svgs.

File: java/server/src/org/openqa/selenium/grid/console/src/css/theme.css (hunk 14 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/src/css/theme.css]: Done.

File: java/server/src/org/openqa/selenium/grid/console/.gitignore (hunk 1 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/.gitignore:10]: The global selenium .gitignore file?

File: java/server/src/org/openqa/selenium/grid/console/public/index.html (hunk 18 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/public/index.html]: Removed.

File: java/server/src/org/openqa/selenium/grid/console/public/css/bootstrap.min.css (hunk 4 above)
Comment (AutomatedTester) [reply java/server/src/org/openqa/selenium/grid/console/public/css/bootstrap.min.css:1]: We want the site to work without having to fetch anything over the internet. The UI might be behind firewalls and might not have access to the internet

File: java/server/src/org/openqa/selenium/grid/console/public/css/fontawesome.min.css (hunk 5 above)
Comment (AutomatedTester) [reply java/server/src/org/openqa/selenium/grid/console/public/css/fontawesome.min.css:1]: We want the site to work without having to fetch anything over the internet. The UI might be behind firewalls and might not have access to the internet

File: java/server/src/org/openqa/selenium/grid/console/package-lock.json (hunk 3 above)
Comment (AutomatedTester) [reply java/server/src/org/openqa/selenium/grid/console/package-lock.json:1]: It's required.

File: java/server/src/org/openqa/selenium/grid/console/public/index.html (hunk 19 above)
Comment (phanirithvij) [reply java/server/src/org/openqa/selenium/grid/console/public/index.html:8]: Added selenium logos

File: java/server/src/org/openqa/selenium/grid/console/.gitignore (hunk 1 above)
Comment (AutomatedTester) [reply java/server/src/org/openqa/selenium/grid/console/.gitignore:10]: yes please
